### PR TITLE
[#5051] GravitinoClient needs metalake to obtain version

### PR DESCRIPTION
**Pull request by lsyulong**
What changes were proposed in this pull request?
[Improvement] GravitinoClient needs metalake to obtain version

Why are the changes needed?

Fix: #5051 

Does this PR introduce any user-facing change?
No

### CODE REVIEW

**Comment by justinmclean** [comment]
Thanks for your contribution. There are a couple of unnecessary changes to the documentation here, can you please fix those? It also looks like you using an old branch, it would be best to use a new branch off the main one.

**Comment by lsyulong** [comment]
> Thanks for your contribution. There are a couple of unnecessary changes to the documentation here, can you please fix those? It also looks like you using an old branch, it would be best to use a new branch off the main one.感谢您的贡献。这里的文档有一些不必要的更改，您能修复这些吗？它看起来也像您使用旧分支，最好使用主分支之外的新分支。

Hello, I didn't quite understand what it meant. Can you explain it to me

**Comment by justinmclean** [comment]
Your PR include changes to apache-hive-catalog.md and a couple of other .md documentation files, these changes are not related to the issue and should not be in the PR. Your branch is also 3 months out of date, it would be best to base your PR on a newer branch.

**Comment by lsyulong** [comment]
> Your PR include changes to apache-hive-catalog.md and a couple of other .md documentation files, these changes are not related to the issue and should not be in the PR. Your branch is also 3 months out of date, it would be best to base your PR on a newer branch.您的 PR 包括对 apache-hive-catalog.md 和其他几个 .md 文档文件的更改，这些更改与问题无关，不应出现在 PR 中。你的分支也已经过期 3 个月了，最好将你的 PR 建立在较新的分支上。

Hello, I am unable to use a specific commit file specified in the submitted PR. Can you help me

File: clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java [hunk 1]
```diff
@@ -537,7 +555,7 @@ public GravitinoClient build() {
           metalakeName != null && !metalakeName.isEmpty(),
           "The argument 'metalakeName' must be a valid name");
```
Comment (justinmclean) [inline clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java:556]: I think the checks on metalakeName will stop the new constructor from being called if metalake name is null or empty.

File: clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java [hunk 2]
```diff
@@ -84,6 +84,24 @@ private GravitinoClient(
...
+  /**
+   * Constructs a new GravitinoClient with the given URI, authenticator and AuthDataProvider.
+   *
+   * @param uri The base URI for the Gravitino API.
+   * @param authDataProvider The provider of the data which is used for authentication.
+   * @param checkVersion Whether to check the version of the Gravitino server. Gravitino does not
+   *     support the case that the client-side version is higher than the server-side version.
+   * @param headers The base header for Gravitino API.
+   * @throws NoSuchMetalakeException if the metalake with specified name does not exist.
+   */
```
Comment (justinmclean) [inline clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java:96]: How is the metalake set after calling this constructor?

Comment (lsyulong) [reply clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoClient.java:96]: Hello, I will close this PR first and make some modifications. Later, I will open a new PR for further discussion

**Pull request closed without merging**